src/nnet3/decodable-batch-looped.h

```diff
@@ -0,0 +1,357 @@
+    opts->Register("extra-left-context-initial", &extra_left_context_initial,
+                   "Extra left context to use at the first frame of an utterance (note: "
+                   "this will just consist of repeats of the first frame, and should not "
+                   "usually be necessary.");
+    opts->Register("frame-subsampling-factor", &frame_subsampling_factor,
+                   "Required if the frame-rate of the output (e.g. in 'chain' "
+                   "models) is less than the frame-rate of the original "
+                   "alignment.");
+    opts->Register("max-batch-size", &max_batch_size,
+                   "number of max sequences for decodable.");
+    opts->Register("compute-interval", &compute_interval,
+                   "how many microseconds to wait after one computation.");
+    opts->Register("acoustic-scale", &acoustic_scale,
+                   "Scaling factor for acoustic log-likelihoods");
+    opts->Register("frames-per-chunk", &frames_per_chunk,
+                   "Number of frames in each chunk that is separately evaluated "
+                   "by the neural net.  Measured before any subsampling, if the "
+                   "--frame-subsampling-factor options is used (i.e. counts "
+                   "input frames.  This is only advisory (may be rounded up "
+                   "if needed.");
+    opts->Register("debug-computation", &debug_computation, "If true, turn on "
+                   "debug for the actual computation (very verbose!)");
+
+    // register the optimization options with the prefix "optimization".
+    ParseOptions optimization_opts("optimization", opts);
+    optimize_config.Register(&optimization_opts);
+
+    // register the compute options with the prefix "computation".
+    ParseOptions compute_opts("computation", opts);
+    compute_config.Register(&compute_opts);
+  }
+};
+
+
+/**
+   When you instantiate class DecodableNnetBatchLooped, you should give it
+   a const reference to this class, that has been previously initialized.
+ */
+class DecodableNnetBatchLoopedInfo  {
+ public:
+  // The constructor takes a non-const pointer to 'nnet' because it may have to
+  // modify it to be able to take multiple iVectors.
+  DecodableNnetBatchLoopedInfo(const NnetBatchLoopedComputationOptions &opts,
+                               Nnet *nnet);
+
+  // This constructor takes the priors from class AmNnetSimple (so it can divide by
+  // them).
+  DecodableNnetBatchLoopedInfo(const NnetBatchLoopedComputationOptions &opts,
+                               AmNnetSimple *nnet);
+
+  // this constructor is for use in testing.
+  DecodableNnetBatchLoopedInfo(const NnetBatchLoopedComputationOptions &opts,
+                               const Vector<BaseFloat> &priors,
+                               Nnet *nnet);
+
+  void Init(const NnetBatchLoopedComputationOptions &opts,
+            Nnet *nnet);
+
+  const NnetBatchLoopedComputationOptions &opts;
+
+  const Nnet &nnet;
```

Comment:
Members of a class end with an underscore in general.
I would recommend changing this class to a struct and remove `public`.

Reply:
mm I think this was copied from something else.
This class appears to be 'struct-like', i.e. it is mostly just visible members, but has a few functions for initialization.
So I think we decided the members were part of the public interface.

Reply:
ok